### ISSUE TITLE
0.4 Tracking Issue (Bevy 0.8 Update)

### ISSUE THREAD

**Issue by JonahPlusPlus** [JonahPlusPlus/bevy_atmosphere#18]
Edit: Changed this to a tracking issue.

0.4 update changes:

- [x] Update dependencies to bevy 0.8 compatible versions
- [x] Rename `AtmosphereMat` to `Atmosphere`
- [x] Rewrite material creation
- [x] Rewrite shaders in WGSL
  - [x] Create exports for shader module
  - [x] ~~Default feature gate plugin and material resource~~
  - [x] Add noise via hash function to remove banding (been a while, but I think this was an issue)
- [x] Use texture target and update texture when resource updates
- [x] Update LICENSE to be dual-licensed under both MIT and Apache-2.0 (optional, needs consent from prior contributors)
- [x] Update example
- [x] ~~Feature gate dynamic~~

**Comment by Shatur** [comment]
Sounds nice! Have a good vacation!

**Comment by JonahPlusPlus** [comment]
Thanks, will do!

**Comment by JonahPlusPlus** [comment]
Everything here is done, but I still have some changes to make tomorrow, so it will probably be released Monday afternoon.

Dithering appears as chromatic aberration during dawn and dusk, when there is huge contrast. I may try to find another method or just increase the default resolution from 512 to 1024 to hide the problem (everytime I x2 the size, the time to render x4, so I can't go overboard).

I also got to write docs and examples. I also have an article about the internals in the works that I would like to get done.

**Comment by JonahPlusPlus** [comment]
So I thought wrong and it may take a little longer, but it will be worth it. You see, I was prepared to let dynamic resolution go, and just bake a resolution of 512px into the crate. However, I figured out a solution that allows for changing the resolution of the sky box texture at runtime. This means that games could have settings for their players to trade between quality and performance, by increasing or decreasing the resolution.

However, there was an issue at lower resolutions that made the edges of the skybox visible. I believe a cube texture is the solution, since it can handle texture filtering better than a plain 2D texture. I spent a bunch of time getting the compute and fragment shaders to use two separate views of the same texture, but I have yet to correct the orientations of the mesh faces, so I'm not sure yet if it is fixed or not.
